### PR TITLE
Introduce delegated user feature

### PR DESCRIPTION
**Pull request by maartenba**
This PR introduces a "delegated user" role.

Delegated users:
- Are not allowed to push packages
- Can not create a new API key
- Can not modify package owners
- Can be enforced to require login via a specific external account provider. In addition, if external account provider is required for a delegated user this same setting applies to any user that co-owns a package.

For example, the `microsoft` user on NuGet.org:
- Is not allowed to push packages
- Can not create a new API key
- Can not modify package owners
- Is enforced to require login via a specific external account provider
- Any co-owner for packages owned by `microsoft` requires login via a specific external account provider

Work in progress, but feel free to give it a pass @skofman1 @xavierdecoster @yishaigalatzer @blowdart 

(note that this PR is a very much reduced version of the spec described here: https://github.com/NuGet/Home/wiki/%5Bspec%5D-Support-organizational-accounts)

### CODE REVIEW

**Comment by skofman1** [comment]
Can you please provide more context on scenarios involving "delegate user"? For example, how do we push a new package with "Microsoft" owner?

**Comment by maartenba** [comment]
Not. Only users should be able to (e.g. The CI user push a package, then adds microsoft as the owner. Newer versions are always pushed by this CI user)

Goal is to have microsoft user as an account that can not mutate data.

**Comment by skofman1** [comment]
Can anyone add Microsoft as an co-owner to his package? How do we enforce that only authorized users are allowed to do that? I'm asking about the user scenario, not technical aspects.

**Comment by maartenba** [comment]
Everyone can, but Microsoft account has to acknowledge ownership (the same flow we have today)

File: src/NuGetGallery/Strings.resx [hunk 1]
```diff
@@ -380,4 +380,10 @@ The {2} Team</value>
   <data name="AzureActiveDirectory_Caption" xml:space="preserve">
     <value>Azure Active Directory</value>
   </data>
+  <data name="ApiKeyNotResetForDelegatedUser" xml:space="preserve">
+    <value>The API key has not been updated. The API key can not be edited for delegated users.</value>
+  </data>
+  <data name="CantAddApiKeyCredentialCanNotBeAdedForUserInDelegatedRole" xml:space="preserve">
+    <value>API key credential can not be aded for user in delegated role.</value>
```
Comment (scottbommarito) [inline src/NuGetGallery/Strings.resx]: "added" (typo)

**Comment by maartenba** [comment]
Wondering if anything will still hapen with this (and the [spec](https://github.com/NuGet/Home/wiki/%5Bspec%5D-Support-organizational-accounts)) or if it can be closed? 

From recent code changes it seems the approach I took in this PR is less interesting going forward, so closing makes sense.  /cc @skofman1

**Comment by skofman1** [comment]
Thanks for reminding about this PR @maartenba ! Organizations is still on the backlog, and we will tackle it in the not distant future. However, as you said, the approach will be different from the current spec. Closing this PR.

**Pull request closed without merging**